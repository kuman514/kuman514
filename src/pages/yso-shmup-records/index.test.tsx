import { render } from '@testing-library/react';

import YSOShmupRecordsPage from '.';

describe('YSOShmupRecordsPage', () => {
  it('should show as expected', () => {
    const { container } = render(<YSOShmupRecordsPage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
