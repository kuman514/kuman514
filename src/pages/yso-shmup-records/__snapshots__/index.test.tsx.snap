// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YSOShmupRecordsPage should show as expected 1`] = `
<main
  class="min-h-screen flex flex-col px-8 py-24 gap-16 sm:px-24 max-w-6xl"
>
  <div
    class="w-full text-right"
  >
    최근 수정 일자: 
    2024
    년 
    9
    월
     
    8
    일
  </div>
  <h1
    class="text-bold text-4xl sm:text-5xl md:text-6xl"
  >
    YSOShmupRecords
  </h1>
  <div
    class="w-full flex flex-row flex-wrap justify-center items-center gap-4"
  >
    <img
      alt="YSOShmupRecords의 기록 리스트"
      class="w-fit max-h-80 object-contain"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="YSOShmupRecords의 기록 아티클"
      class="w-fit max-h-80 object-contain"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <img
      alt="YSOShmupRecords의 아카트로닉스 오시는 길"
      class="w-fit max-h-80 object-contain"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </div>
  <section
    class="flex flex-col gap-8 undefined"
  >
    <h2
      class="text-bold text-2xl sm:text-3xl md:text-4xl"
    >
      개요
    </h2>
    <section>
      <ul
        class="list-disc pl-8"
      >
        <li>
          개인 프로젝트
        </li>
        <li>
          구현 내용
          <ul
            class="list-disc pl-8"
          >
            <li>
              프로젝트 세팅, 태스크 브레이크다운, 디자인, 퍼블리싱, 기능 구현, 테스트, 배포, 문서 및 업데이트 로그 작성, 유지보수를 담당함.
            </li>
            <li>
              react-router-dom과 Vercel Rewrite를 사용하여 각 페이지 라우팅.
            </li>
            <li>
              기록 열람, 외부로 공유, 다크 모드 등등 각종 기능 구현.
            </li>
            <li>
              Vite + Yarn Berry로 프로젝트를 세팅하여 Create React App 대비 로딩 시간 30초 → 5초로 감소, 설치 속도가 5배 빨라짐.
            </li>
            <li>
              데이터를 가져오는 동안 UI가 멈춰있어 보이지 않기 위해 스켈레톤 컴포넌트 적용.
            </li>
            <li>
              Lambda + DynamoDB + API Gateway + Node.js를 이용하여 기록 등의 데이터를 제공하고 관리하는 서버리스 백엔드 및 백오피스 구축.
            </li>
          </ul>
        </li>
      </ul>
    </section>
  </section>
  <section
    class="flex flex-col gap-8 undefined"
  >
    <h2
      class="text-bold text-2xl sm:text-3xl md:text-4xl"
    >
      방문하기
    </h2>
    <section>
      [
       
      <a
        class="text-blue-500 hover:text-purple-500 undefined"
        href="https://github.com/kuman514/YSOShmupRecords"
        target="_blank"
      >
        GitHub 저장소
      </a>
       
      ] [
       
      <a
        class="text-blue-500 hover:text-purple-500 undefined"
        href="https://yso-shmup-records.vercel.app/"
        target="_blank"
      >
        사이트
      </a>
       
      ]
    </section>
  </section>
  <section
    class="flex flex-col gap-8 undefined"
  >
    <h2
      class="text-bold text-2xl sm:text-3xl md:text-4xl"
    >
      사용한 기술 스택
    </h2>
    <section
      class="flex flex-row flex-wrap gap-2"
    >
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        React
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        TypeScript
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Vite
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Yarn Berry
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Vercel
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Axios
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Prettier
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Figma
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Lambda
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        DynamoDB
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        API Gateway
      </span>
      <span
        class="font-mono bg-gray-300 text-red-700 text-sm px-1 py-0.5 rounded undefined"
      >
        Node.js
      </span>
    </section>
  </section>
  <section
    class="flex flex-col gap-8 undefined"
  >
    <h2
      class="text-bold text-2xl sm:text-3xl md:text-4xl"
    >
      스켈레톤 컴포넌트 적용
    </h2>
    <section
      class="flex flex-col gap-4"
    >
      <div
        class="flex justify-center items-center"
      >
        <img
          alt="YSOShmupRecords의 기록 아티클 스켈레톤 컴포넌트"
          class="max-h-80 object-contain"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <p
        class="text-base font-normal indent-4 undefined"
      >
        <strong
          class="bg-emphasize-underline from-transparent from-70% to-green-500 dark:to-green-600 to-70% undefined"
        >
          스켈레톤 컴포넌트는 데이터를 가져오는 동안 임시 콘텐츠를 표시하는 컴포넌트
        </strong>
        입니다. 사용자는 콘텐츠를 기다리다가 쉽게 지치고 지루함을 느끼므로, 데이터를 가져오는 동안 단순히 빈 화면만 보여준다면 많은 사람들은 사이트를 떠나게 될 것입니다. 그래서 이를 방지하기 위해,
        <strong
          class="bg-emphasize-underline from-transparent from-70% to-green-500 dark:to-green-600 to-70% undefined"
        >
          애니메이션 효과가 들어간 스켈레톤 컴포넌트를 직접 만들어, 기록 목록이나 기록 아티클 로딩 등에 적용
        </strong>
        하였습니다.
      </p>
    </section>
  </section>
  <section
    class="flex flex-col gap-8 undefined"
  >
    <h2
      class="text-bold text-2xl sm:text-3xl md:text-4xl"
    >
      서버리스 백엔드 구축 및 연동
    </h2>
    <section
      class="flex flex-col gap-4"
    >
      <div
        class="w-full flex flex-row flex-wrap justify-center items-center gap-4"
      >
        <img
          alt="YSOShmupRecords의 백오피스 Writronics의 기록 아티클"
          class="w-fit max-h-80 object-contain"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="YSOShmupRecords의 백오피스 Writronics의 기록 리스트"
          class="w-fit max-h-80 object-contain"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <img
          alt="YSOShmupRecords의 백오피스 Writronics의 기록 수정 화면"
          class="w-fit max-h-80 object-contain"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </div>
      <p
        class="text-base font-normal indent-4 undefined"
      >
        기록 등의 데이터를 더욱 쉽게 관리하기 위해, Lambda + DynamoDB + API Gateway를 이용하여 서버리스 백엔드를 구축하여, YSOShmupRecords와 그의 백오피스 Writronics와 연동하였습니다. 더 나아가, S3 퍼블릭 버킷을 추가적으로 구축하여, Writronics에서 이미지 파일 직접 업로드가 가능하게끔 만들었습니다. 백엔드 상에서, 기록 목록과 기록 아티클 등의 열람은 누구나 할 수 있지만 추가/편집/삭제는 관리자 로그인이 필요하도록 만들었습니다.
      </p>
    </section>
  </section>
</main>
`;
